mqt/lints.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import subprocess
+from os.path import join
+import pylint.lint
+
+from . import getaddons
+
+package_path = os.path.dirname(os.path.realpath(__file__))
+
+
+def get_count_fails(linter_stats, msgs_no_count=None):
+    """Verify the dictionary statistics to get number of errors.
+
+    :param linter_stats: Dict of type pylint.lint.Run().linter.stats
+    :param msgs_no_count: List of messages that will not add to the failure count.
+    :return: Integer with quantity of fails found.
+    """
+    return sum([
+        linter_stats['by_msg'][msg]
+        for msg in linter_stats['by_msg']
+        if msg not in msgs_no_count])
+
+
+def get_sub_paths(paths):
+    """Get list of subdirectories
+    if `__init__.py` file not exists in root path then
+    get subdirectories.
+    Why? More info here:
+        https://www.mail-archive.com/code-quality@python.org/msg00294.html
+
+    :param paths: List of paths
+    :return: Return list of paths with subdirectories.
+    """
+    subpaths = []
+    for path in paths:
+        if not os.path.isfile(os.path.join(path, '__init__.py')):
+            subpaths.extend(
+                [os.path.join(path, item)
+                 for item in os.listdir(path)
+                 if os.path.isfile(os.path.join(path, item, '__init__.py')) and
+                 (not getaddons.is_module(os.path.join(path, item)) or
+                  getaddons.is_installable_module(os.path.join(path, item)))])
+        else:
+            if not getaddons.is_module(path) or \
+                    getaddons.is_installable_module(path):
+                subpaths.append(path)
+    return subpaths
+
+
+def run_pylint(paths, cfg,
+               beta_msgs=None,
+               sys_paths=None,
+               extra_params=None):
+    """Execute pylint command from original python library
+    
+    :param paths: List of paths of python modules to check pylint
+    :param cfg: String name of pylint configuration file
+    :param sys_paths: List of paths to append to sys path
+    :param extra_params: List of parameters extra to append
+        in pylint command
+    :return: Dict with python linter stats
+    """
+    if sys_paths is None:
+        sys_paths = []
+    if extra_params is None:
+        extra_params = []
+    if not cfg:
+        cfg = join(package_path, 'cfg/pylint.cfg')
+    sys.path.extend(sys_paths)
+    cmd = ['--rcfile=' + cfg]
+    cmd.extend(extra_params)
+    sub_paths = get_sub_paths(paths)
+    if not sub_paths:
+        raise UserWarning("Python modules not found in paths"
+                          " {paths} ".format(paths=paths))
+    cmd.extend(sub_paths)
+    pylint_res = pylint.lint.Run(cmd, exit=False)
+    return pylint_res.linter.stats
+
+
+def flake():
```

Comment:
@nhomar IMO it would be better to split each sub command in its own module.
pylint -> pylint.py or ???.py
flake8 -> flake8.py or ???.py

Reply:
Agree. Maybe:
```
linters/common.py (helper methods here)
linters/pylint.py
linters/flake8.py
```

Reply:
flake is so so simple (just 10 lines) it was separated before and I just join them because the name of the file is "lints".

I do not see too much value have 2 files for that and a sub folder.

Remember that ones something will become big enough then may be extract from here will be proposed.

but I will considere this.

Reply:
@nhomar I prefer to keep thing separate from the start. Once you split a module, you lost the 
change histrory or at least it become more difficul to follow it. Moreover it's also to stay consistent with what we ask for odoo addons. One module by model. (it's just my POV).